kong/plugins/prometheus/exporter.lua

```diff
@@ -413,13 +451,15 @@ local function metric_data(write_fn)
   end
 
   -- only export upstream health metrics in traditional mode and data plane
-  if role ~= "control_plane" and should_export_upstream_health_metrics then
+  if role ~= "control_plane" and should_export_upstream_health_metrics and ngx_now() - current > ITERATE_UPSTREAM_HEALTH_METRICS_INTERVAL then
     -- erase all target/upstream metrics, prevent exposing old metrics
+    current = ngx_now()
     metrics.upstream_target_health:reset()
 
     -- upstream targets accessible?
     local upstreams_dict = get_all_upstreams()
     for key, upstream_id in pairs(upstreams_dict) do
+      yield()
```

Comment:
```suggestion
      yield(true)
```

To avoid excessive yielding.